### DISCUSSION TITLE
How to remove SQL double brackets --dev mode?

### DISCUSSION THREAD

**Discussion by MengLinMaker** [pocketbase/pocketbase#4401] [Q&A]
I noticed in --dev mode, PocketBase would print SQL statements with double brackets:
```
CREATE TABLE {{_migrations}} (
    [[file]] VARCHAR(255) PRIMARY KEY NOT NULL,
    [[applied]] INTEGER NOT NULL
  );
```
Unfortunately, this SQL is not compatible with sqlite, due to the double brackets. 

Now, I could use regex to remove the brackets in a hacky way.
Is there a better solution?

For context, I'm trying to stream SQL statements to replicate the database on Turso.

**Comment by ganigeorgiev** [comment]
The logged statements are only for dev purposes and they are not intended to be executed as raw SQL (_some of the printed values could be also shorted/redacted in case blob or other similar data is printed_).

> For context, I'm trying to stream SQL statements to replicate the database on Turso.

Could you elaborate why?
I have no practical experience with Turso and it seems to be fairly popular based on https://github.com/pocketbase/pocketbase/discussions/2638 but I don't understand what would be the benefit of combining it with PocketBase.
If you need only backup replication you can try Litestream.

**Reply by ganigeorgiev** [reply]
Also note that with the [planned refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) there will be support for loading a custom SQLite build or driver compatible with `database/sql` so you'll be able to connect the Turso/libsql driver if you really need to (_technically you can do that already with the `//go:linkname` directive and replacing `core.connectDB` function but I don't recommend it_).

**Reply by MengLinMaker** [reply]
My ultimate aim is to create a low management and cost SaaS boilerplate that does not compromise on scalability.

Turso can be used to scale PocketBase with read replicas, plus placing databases closer to users to improve performance.

`Litestream` cannot scale to multiple servers, otherwise the data would be corrupted.

I am aware that there are other solutions like `marmot`, `rqlite`, `dqlite` which use distributed consensus to replicate. However, I'm concerned with the loss on write performance, which is already a bottleneck of sqlite. Plus they have their own restrictions and add management complexity.

Do consider that I'm mainly using PocketBase for authentication. So maybe something like [Lucia](https://github.com/lucia-auth/lucia) is also worth considering.

At this point I'm also realising that sqlite complicates scaling since compute needs to scale with database storage.

**Reply by ganigeorgiev** [reply]
> Turso can be used to scale PocketBase with read replicas, plus placing databases closer to users 

Both use cases are not suitable with PocketBase because out of the box PocketBase can scale only vertically, aka. on a single server.

Having the "databases closer to users" doesn't matter if your backend application can run only on a single server as it is with PocketBase.

As mentioned in the other similar discussions, if you need HA or horizontal scaling you might be better with a custom tailored solution or if you want to use BaaS look at Supabase, Appwrite, Nhost, SurrealDB, EdgeDB, etc.

**Reply by MengLinMaker** [reply]
Maybe some parts of PocketBase cannot scale horizontally.

But I'm only interested in the oauth2 part. Surely that part is horizontally scalable.
An alternative is to use Lucia. Might be a better choice.

I'll still create a rough prototype integrating PocketBase and Turso in case that helps anyone.
